odxtools/admindata.py

```diff
@@ -46,28 +66,133 @@ class Modification:
     change: Optional[str] = None
     reason: Optional[str] = None
 
+    @staticmethod
+    def from_et(et_element: ElementTree.Element,
+                doc_frags: List[OdxDocFragment]) \
+            -> "Modification":
+        change = et_element.findtext("CHANGE")
+        reason = et_element.findtext("REASON")
+
+        return Modification(change=change,
+                            reason=reason)
+
+@dataclass()
+class CompanyRevisionInfo:
+    company_data_ref: OdxLinkRef
+    revision_label: Optional[str] = None
+    state: Optional[str] = None
+
+    _company_data: Optional[CompanyData] = None
+    @property
+    def company_data(self) -> CompanyData:
+        assert self._company_data is not None
```

Comment:
Avoid throwing in getters, user of API always assume that getters have no logic

See also https://stackoverflow.com/questions/48778945/by-design-should-a-property-getter-ever-throw-an-exception-in-python

Reply:
hm, the problem with this is that this would make the type of the return value `Optional[T]` thus making mypy happy would become pretty awkward for mandatory items that are defined by reference. At this particular location, the `company_data` is supposed to be optional anyway because the reference can be `None` as well. That said, I still think that raising an `AssertationError` is okay to do even if it's not expected by the user because this represents an error in the program logic to call such properties before the references are resolved. (In e.g. C++, this usually leads to an `abort()` or a segfault...)

Reply:
ok, scrap my last comment: for COMPANY-REVISION-INFO the COMPANY-DATA-REF tag *is* indeed mandatory. what to do about it?

Reply:
if you must throw, at least throw something that could be caught (one of the exception classes defined in odxtools) and add an error message

Reply:
ok. separate PR?!

Reply:
BTW: I think that it is possible to catch `AssertationError` and I also think that this exception is appropriate at this location because it signifies a programming error by the user. (For most of the asserts this is not really the case, in particular not those who's primary purpose is to satisfy mypy...)